File: exercises/exercism.io/clojure/anagram/anagram.clj
(ns anagram
  (:use [clojure.string :only [lower-case]]))

(defn anagrams-for [word candidates]
  (letfn [(keify [string] (sort (lower-case string)))
          (anagram? [candidate]
            (and (= (keify word) (keify candidate)) (not= word candidate)))]
    (filter anagram? candidates)))

File: exercises/exercism.io/ruby/anagram/anagram.rb
class Anagram

  def initialize(word)
    @word = word
  end

  def match(candidates)
    candidates.select do |candidate|
      different_word?(candidate) && anagram?(candidate)
    end
  end

 private
 
  def anagram?(another_word)
    sorted_chars(@word) == sorted_chars(another_word)
  end

  def sorted_chars(string)
    string.downcase.chars.sort
  end

  def different_word?(another_word)
    @word.casecmp(another_word) != 0
  end
  
end
